sdk/spring/azure-spring-cloud-stream-binder-servicebus-queue/src/main/java/com/azure/spring/servicebus/stream/binder/ServiceBusQueueHealthIndicator.java

```diff
@@ -0,0 +1,47 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+package com.azure.spring.servicebus.stream.binder;
+
+import com.azure.spring.integration.servicebus.health.Instrumentation;
+import com.azure.spring.integration.servicebus.health.InstrumentationManager;
+import com.azure.spring.integration.servicebus.queue.ServiceBusQueueTemplate;
+import org.springframework.boot.actuate.health.AbstractHealthIndicator;
+import org.springframework.boot.actuate.health.Health;
+
+/**
+ * Implementation of a {@link AbstractHealthIndicator} returning status information for
+ * service bus queue.
+ */
+public class ServiceBusQueueHealthIndicator extends AbstractHealthIndicator {
```

Comment:
Why do we extend AbstractHealthIndicator instead of implementing HealthIndicator interface?